### ISSUE TITLE
Not all tensors on same device

### ISSUE THREAD

**Issue by Squeemos** [aai-institute/tianshou#589]
- [x] I have marked all applicable categories:
    + [x] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
0.4.7 0.18.3 1.10.2+cu113 1.21.5 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] win32

I'm using my own custom environment (which is very simple) but it returns it's values as numpy arrays. I'm using the sample on the [main page](https://github.com/thu-ml/tianshou/)

```python
gym.envs.registration.register(
        id="VacuumLand-v0",
        entry_point=VacuumLand,
        max_episode_steps=100,
        kwargs={'height' : 10, 'width' : 10, 'trash' : 10, 'as_image' : False, 'penalty' : True, 'seed' : 0}
    )

task = "VacuumLand-v0"
lr, epoch, batch_size = 1e-3, 10, 64
train_num, test_num = 10, 10
gamma, n_step, target_freq = 0.9, 10, 320
buffer_size = 20000
eps_train, eps_test = 0.1, 0.05
step_per_epoch, step_per_collect = 10000, 10
logger = ts.utils.TensorboardLogger(torch.utils.tensorboard.SummaryWriter('log/dqn'))  # TensorBoard is supported!

train_envs = ts.env.DummyVectorEnv([lambda: gym.make(task) for _ in range(train_num)])
test_envs = ts.env.DummyVectorEnv([lambda: gym.make(task) for _ in range(test_num)])

env = gym.make(task)
net = ts.utils.net.common.Net(state_shape=env.observation_space.shape, action_shape=env.action_space.n, hidden_sizes=[64,64], device="cuda")
optim = torch.optim.Adam(net.parameters(), lr=lr)

policy = ts.policy.DQNPolicy(net, optim, gamma, n_step, target_update_freq=target_freq)
train_collector = ts.data.Collector(policy, train_envs, ts.data.VectorReplayBuffer(buffer_size, train_num), exploration_noise=True)
test_collector = ts.data.Collector(policy, test_envs, exploration_noise=True)

result = ts.trainer.offpolicy_trainer(
    policy, train_collector, test_collector, epoch, step_per_epoch, step_per_collect,
    test_num, batch_size, update_per_step=1 / step_per_collect,
    train_fn=lambda epoch, env_step: policy.set_eps(eps_train),
    test_fn=lambda epoch, env_step: policy.set_eps(eps_test),
    stop_fn=lambda mean_rewards: mean_rewards >= 1.0,
    logger=logger)
print(f'Finished training! Use {result["duration"]}')
```

I'm running into an error that I can't seem to find on my own where the device isn't the same for the network and the observations. Should I use a wrapper for the environment or should I be passing an arg somewhere where I'm not?

**Comment by Trinkle23897** [comment]
```diff
  net = ts.utils.net.common.Net(state_shape=env.observation_space.shape, action_shape=env.action_space.n, hidden_sizes=[64,64], device="cuda")
+ net = net.to("cuda")
```

Please refer to the full script at https://github.com/thu-ml/tianshou/blob/f13e415eb0de55baca5dc0d6fae39d6a38e8bc0b/test/discrete/test_dqn.py#L77-L83

**Comment by Squeemos** [comment]
Aha thanks so much! Knew I was missing something.
